core/src/mindustry/creeper/CreeperUtils.java

```diff
@@ -203,8 +201,8 @@ public static void init(){
     public static void depositCreeper(Tile tile, float radius, float amount){
         Geometry.circle(tile.x, tile.y, (int) radius, (cx, cy) -> {
             Tile ct = world.tile(cx, cy);
-            if(!validTile(ct) || (tile.block() instanceof StaticWall || (tile.floor() != null && !tile.floor().placeableOn || tile.floor().isDeep() || tile.block() instanceof Cliff)))
-                return;
+            // don't add creeper to staticwalls, cliffs, space or deep water
+            if(!validTile(ct) || !(ct.breakable() || ct.block().alwaysReplace || ct.block() instanceof TreeBlock) || ct.floor().placeableOn || ct.floor().isDeep()) return;
```

Comment:
how about putting this in a seperate function called canDeposit ? that way it can be used in multiple instances without copy & paste